### PR TITLE
Adding InputPreset::VoicePerformance. 

### LINKED ISSUE #554: Add InputPreset::VoicePerformance to builder
A new InputPreset was added in Q:  AAUDIO_INPUT_PRESET_VOICE_PERFORMANCE

We need to add this preset (10) to Oboe. If version < Q then force it to VOICE_RECOGNITION.

See also https://github.com/google/oboe/issues/552

### PR DESCRIPTION
**Pull request by atneya**
Fixes #554. Need to investigate impacts of adding as default.

### CODE REVIEW

File: include/oboe/Definitions.h [hunk 1]
```diff
@@ -346,6 +346,11 @@ namespace oboe {
      * This attribute only has an effect on Android API 28+.
      */
     enum InputPreset : int32_t { // aaudio_input_preset_t
+        /**
+         * This is for Oboe, it should not be passed to AAudio.
+         */
+        Unspecified = -1, // No corresponding AAudio preset
```
Comment (dturner) [inline include/oboe/Definitions.h]: Maybe change to `Unspecified = kUnspecified` - check with jean-michel

**Comment by atneya** [comment]
Add extra conversion in OpenSLES.

**Comment by atneya** [comment]
Reverting changes except for updating InputPreset to reflect new AAudio preset value. Changing default to Unspecified may be a possible alternate improvement, but latency impacts need to be investigated.

File: include/oboe/AudioStreamBase.h [hunk 2]
```diff
@@ -160,7 +160,9 @@ class AudioStreamBase {
     Usage                           mUsage = Usage::Media;
     /** Stream content type. Only active on Android 28+ */
     ContentType                     mContentType = ContentType::Music;
-    /** Stream input preset. Only active on Android 28+ */
+    /** Stream input preset. Only active on Android 28+
+     * InputPreset::Unspecified should be considered as a possible default alternative.
```
Comment (philburk) [inline include/oboe/AudioStreamBase.h]: Mark things like this with "TODO". Android Studio will detect them.

**Pull request merged**